sklearn/tests/test_calibration.py

```diff
@@ -275,6 +275,34 @@ def multiclass_brier(y_true, proba_pred, n_classes):
     assert calibrated_brier < 1.1 * uncalibrated_brier
 
 
+def test_calibration_zero_probability():
+    class FakeClassifier():
+        def __init__(self, n_classes):
+            self.classes_ = n_classes
+
+        def predict_proba(self, X):
+            return np.empty_like(X)
+
+    class FakeCalibrator():
+        # This function is called from _CalibratedClassifier.predict_proba.
+        def predict(self, X):
+            return np.zeros_like(X)
```

Comment:
Should the prediction have the same shape as X ? I thought it would be ```np.zeros(X.shape[0])``

Reply:
I think X should always be `(n_samples, 1)` and the output as well. So this looks correct to me if I am not mistaken.

Reply:
Yes, as @ogrisel said, X is always `(n_samples, 1)`, but using `X.shape[0]` is clearer as @jeremiedbb, so I fixed it, thanks.